misc/ballerina-bindgen/src/main/java/org/ballerinalang/bindgen/model/JParameter.java

```diff
@@ -0,0 +1,145 @@
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.bindgen.model;
+
+import java.lang.reflect.Parameter;
+
+import static org.ballerinalang.bindgen.command.BindingsGenerator.getAllJavaClasses;
+import static org.ballerinalang.bindgen.command.BindingsGenerator.setClassListForLooping;
+import static org.ballerinalang.bindgen.utils.BindgenConstants.BALLERINA_STRING;
+import static org.ballerinalang.bindgen.utils.BindgenConstants.BALLERINA_STRING_ARRAY;
+import static org.ballerinalang.bindgen.utils.BindgenUtils.getBallerinaHandleType;
+import static org.ballerinalang.bindgen.utils.BindgenUtils.getBallerinaParamType;
+import static org.ballerinalang.bindgen.utils.BindgenUtils.getPrimitiveArrayType;
+
+/**
+ * Class for storing specific parameter details of a Java method used for Ballerina bridge code generation.
+ */
+public class JParameter {
+
+    private String type;
+    private String externalType;
+    private String shortTypeName;
+    private String componentType;
+    private String fieldName;
+
+    private Boolean isObj = false;
+    private Boolean hasNext = true;
+    private Boolean isString = false;
+    private Boolean isObjArray = false;
+    private Boolean isPrimitiveArray = false;
+
+    JParameter(Class parameterClass) {
+
+        type = parameterClass.getName();
+        shortTypeName = getBallerinaParamType(parameterClass);
+
+        if (!parameterClass.isPrimitive()) {
+            isObj = true;
+        }
+        if (parameterClass.equals(String.class)) {
+            isString = true;
+            shortTypeName = BALLERINA_STRING;
+        } else if (parameterClass.equals(String[].class)) {
+            isString = true;
+            shortTypeName = BALLERINA_STRING_ARRAY;
+        } else {
+            if (!parameterClass.isPrimitive()) {
+                if (parameterClass.isArray()) {
+                    setArrayAttributes(parameterClass);
+                } else {
+                    String paramType = parameterClass.getName();
+                    if (!getAllJavaClasses().contains(paramType)) {
+                        setClassListForLooping(paramType);
+                    }
+                }
+            }
+        }
+        externalType = getBallerinaHandleType(parameterClass);
+        fieldName = "arg";
+    }
+
+    JParameter(Parameter parameter) {
+
+        this(parameter.getType());
+        fieldName = parameter.getName();
+
```

Comment:
extra new line. You also can remove the extra new lines after each method signature as well.

Reply:
Done.